### ISSUE TITLE
Word wrapping duplicates words in strings

### ISSUE THREAD

**Issue by Minoru** [catchorg/Catch2#1400]
### Steps to reproduce

Compile and run the following:

```c++
#define CATCH_CONFIG_MAIN
#include "catch.hpp"

TEST_CASE("Demonstrate the bug in word wrapping") {
    const auto long_string = std::string(
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque nisl \n"
        "massa, luctus ut ligula vitae, suscipit tempus velit. Vivamus sodales, quam in \n"
        "convallis posuere, libero nisi ultricies orci, nec lobortis.\n");

    const auto even_longer_string = long_string + '\n';

    // This will fail; that's intentional.
    REQUIRE(long_string == even_longer_string);
}
```

When ran, this produces the following output:

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
a.out is a Catch v2.4.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
Demonstrate the bug in word wrapping
-------------------------------------------------------------------------------
main.cpp:4
...............................................................................

main.cpp:19: FAILED:
  REQUIRE( long_string == even_longer_string )
with expansion:
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque nisl
  massa, luctus ut ligula vitae, suscipit tempus velit. Vivamus sodales, quam in
  in
  convallis posuere, libero nisi ultricies orci, nec lobortis.
  "
  ==
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque nisl
  massa, luctus ut ligula vitae, suscipit tempus velit. Vivamus sodales, quam in
  in
  convallis posuere, libero nisi ultricies orci, nec lobortis.

  "

===============================================================================
test cases: 1 | 1 failed
assertions: 1 | 1 failed
```

Notice that this output contains "quam in in convallis", whereas the original code only contains "quam in convallis".

### Extra information

* Catch version: 9e1bdca4667295fcb16265eae00efa8423f07007
* Operating System: NixOS 18.03
* Compiler+version: GCC 7.3.0 and Clang 5.0.2

**Comment by JoeyGrajciar** [comment]
After few hours of debugging I found that this is actually issue with clara::TextFlow::Column::iterator formatting with the indentation. In this case the second line has 81 characters which is longer than console width. Basically the problem will occur here 

> auto operator *() const -> std::string {
>               assert( m_stringIndex < m_column.m_strings.size() );
>                assert( m_pos <= m_end );
>               if( m_pos + m_column.m_width < m_end ) // 155 < 155
>                    return addIndentAndSuffix(line().substr(m_pos, m_len));
>                else
>                    return addIndentAndSuffix(line().substr(m_pos, m_end - m_pos)); // m_end - m_pos is 79 or m_column.m_width
>}

Seems that it should be enough to change the condition to `<=` instead of `<`.
With this change the result is:
>"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque nisl
  massa, luctus ut ligula vitae, suscipit tempus velit. Vivamus sodales, quam
  in
  convallis posuere, libero nisi ultricies orci, nec lobortis.
  "
  ==
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque nisl
  massa, luctus ut ligula vitae, suscipit tempus velit. Vivamus sodales, quam
  in
  convallis posuere, libero nisi ultricies orci, nec lobortis.
>
  >"

**Comment by horenmar** [comment]
@JoeyGrajciar Thanks.

Will you open up a PR against [Clara](https://github.com/catchorg/Clara/)?

**Comment by JoeyGrajciar** [comment]
@horenmar Yes i will create one soon :) Probably it will be different as i am not really sure if that condition is needed.

**Comment by horenmar** [comment]
This has been fixed in the dependency, now it needs to find its way upward.

**Comment by Minoru** [comment]
Thank you very much!
